packet.go

```diff
@@ -37,12 +38,18 @@ func New(code Code, secret []byte) *Packet {
 		Secret:     secret,
 	}
 	copy(packet.Authenticator[:], buff[1:])
```

Comment:
I think you could also do

```go
	packet := &Packet{
		Code:       code,
		Identifier: buff[0],
		Secret:     secret,
                Authenticator: buff[1:],
	}
```

Reply:
Your suggestion does not compile. This was original code before the merge request.  Not sure if this can be improved, but a quick google did not pop up anything obvious.

Reply:
ah got it... sorry I missed that Authenticator is an array not a slice. The copy function will drop all bytes that are > 16, so this is good for me :+1: